### ISSUE TITLE
publish failed

### ISSUE THREAD

**Issue by touseefbsb** [nicolasr75/PolymerAspNetCore#1]
I built the project and was able to run it on local host and web api showed value1 and value2. after that I right clicked the project and published it, it started to build and then gave error "Publish Failed".

I checked the error list as per instructions but it was empty, there was no error nor a warning, did I miss any step, before I tried to publish?

**Comment by nicolasr75** [comment]
Try first to build manually. Does that succeed?

Then you may want to change the build logging to Detailed or Diagnostic:
Tools -> Options -> Projects and Solutions -> Build and Run -> MSBuild project build output verbosity

Still no output?

**Comment by touseefbsb** [comment]
yes if I run the project in issexpress normally with visual studio F5, web api runs to show dictionary values. but when I publish, It fails and after doing the detailed build error list setting u mentioned now I get following error in error list, when build fails.

ENOENT: no such file or directory, open 'C:\Users\touseef\source\repos\PolymerAspNetCore\wwwroot\bower_components\polymer\polymer-element.html'

and I can assure that I cannot see any bowercomponents folder in wwwroot folder in the project

**Comment by nicolasr75** [comment]
Ah, ok. The bower components don't get committed to github and they also don't seem to get restored by Visual Studio, probably because the bower.json file is located in wwwroot. You need to update them manually. Open a commandline in wwwroot and do

`bower update`

That should bring you all bower dependencies.

**Comment by touseefbsb** [comment]
ok i have succesfully published and can see the es6-unbundled version of the app in bin folder, now I have a few questions about it.

1. how can I produce other versions, for example bundled es6 or bundled es5 ?
2. how will I actually run this published version of app now? do I still need to do polymer serve?
3. How will this app actually link and work with the backend web api project?

Thanks in advance

**Comment by nicolasr75** [comment]
1. you need to change polymer.json. You can either create your own build configuration or use one of the presets. Read the documentation on polymer-build for more information. Don't forget that a different configuration will create a differently named directory in wwwroot/build and you need to use the same name in the csproj file!

2. see the readme file

3. via the iron-ajax component. See the file my-view1.html

**Comment by touseefbsb** [comment]
To test the published app open a console in the publish directory and execute dotnet PolymerAspNetCore.dll

I am trying to do this but console says the dll doesnt exist in that directory. can you please specify exactly what path should i execute this command in?
i tried executing it in es6-bundled folder, then the build folder, and then in wwwroot folder, it didnt work in anyone of them

**Comment by touseefbsb** [comment]
I was able to serve the app on localhost 5000 port, but when I open it in browser I am getting not found error

**Comment by nicolasr75** [comment]
Use the developper tools in your browser to check which file generates the 404 error. Also check if the publish directory really contains the required files, especially index.html.

**Comment by touseefbsb** [comment]
No the publish directory, i.e bin\release\publishOutput   doesnt have any html file , just some dll files and some json, and a folder with wwwroot, which also has just a json file in it

**Comment by nicolasr75** [comment]
Ok, sorry, I meant the wwwroot directory needs to contain the html files. That you're not having the files means that something went wrong during the publish. I remember that I had similar problems one or two times. 

Could you try to publish again and follow the log output, especially after the line where it says 'polymer-build.cmd'? You should also check the Polymer build directory 'es6-bundled'. If this directory contains all necessary files, then it's a problem of VS not correctly copying the files to the publish wwwroot directory.

**Comment by nicolasr75** [comment]
I just encountered the same problem. My current way around it is to use avsecond 'publish' batch file that copies the Polymer files via xcopy and does not rely on MSBuild doing it. The downside of this is that it has the publish directory hard coded. I will update the code on Github shortly.

**Comment by touseefbsb** [comment]
oh alright thanks
